test/integration/image-optimizer/test/index.test.js

```diff
@@ -0,0 +1,215 @@
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, {})
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"q" parameter (quality) must be a number between 1 and 100`
+    )
+  })
+
+  it('should fail when q is less than 1', async () => {
+    const query = { url: '/vercel-logo.png', w: 64, q: 0 }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, {})
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"q" parameter (quality) must be a number between 1 and 100`
+    )
+  })
+
+  it('should fail when w is 0 or less', async () => {
+    const query = { url: '/vercel-logo.png', w: 0, q: 100 }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, {})
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"w" parameter (width) must be a number greater than 0`
+    )
+  })
+
+  it('should fail when w is not a number', async () => {
+    const query = { url: '/vercel-logo.png', w: 'foo', q: 100 }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, {})
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"w" parameter (width) must be a number greater than 0`
+    )
+  })
+
+  it('should fail when q is not a number', async () => {
+    const query = { url: '/vercel-logo.png', w: 64, q: 'foo' }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, {})
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"q" parameter (quality) must be a number between 1 and 100`
+    )
+  })
+
+  it('should fail when domain is not in next.config.js', async () => {
+    const query = {
+      url: `http://vercel.com/button`,
+      w: 64,
+      q: 100,
+    }
+    const opts = { headers: { accept: 'image/webp' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(`"url" parameter is not allowed`)
+  })
+
+  it('should fail when width is not in next.config.js', async () => {
+    const query = { url: '/vercel-logo.png', w: 1000, q: 100 }
+    const opts = { headers: { accept: 'image/webp' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(400)
+    expect(await res.text()).toBe(
+      `"w" parameter (width) of 1000 is not allowed`
+    )
+  })
+
+  it('should resize relative url', async () => {
+    const query = { url: '/vercel-logo.png', w: 64, q: 100 }
+    const opts = { headers: { accept: 'image/webp' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('content-type')).toBe('image/webp')
+  })
+
+  it('should resize absolute url from localhost', async () => {
+    const query = {
+      url: `http://localhost:${appPort}/vercel-logo.png`,
+      w: 64,
+      q: 100,
+    }
+    const opts = { headers: { accept: 'image/webp' } }
```

Comment:
We should also check that it falls back to the original format if `WebP` isn't supported right?

Reply:
I added fallbacks for `image/png` and `image/jpeg` and added a couple tests 👍